src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -132,6 +132,21 @@ public bool Equals(LockFile other)
                 && LogsEqual(other.LogMessages);
         }
 
+        public LockFile Clone()
+        {
+            var lockFile = new LockFile();
+            lockFile.Version = Version;
+            lockFile.Path = Path;
+            lockFile.ProjectFileDependencyGroups = ProjectFileDependencyGroups?.Select(item => item.Clone()).ToList();
+            lockFile.Libraries = Libraries?.Select(item => item.Clone()).ToList();
+            lockFile.Targets = Targets?.Select(item => item.Clone()).ToList();
+            lockFile.PackageFolders = new List<LockFileItem>(PackageFolders);
+            lockFile.PackageSpec = PackageSpec.Clone();
+            lockFile.LogMessages = new List<IAssetsLogMessage>(LogMessages);
```

Comment:
Same here, this is not really being cloned.

Reply:
fixed